src/other/Opportunity.js

```diff
@@ -0,0 +1,381 @@
+    Opportunity.prototype._class += " graph_Opportunity";
+    Opportunity.prototype.publish("id", "id", "set", "Id for label in Opportunity", function() { return this.getIds(); }, { tags: ["Basic", "Shared"]});
+    Opportunity.prototype.publish("previousGroup", "", "set", "label in Opportunity", function() { return this.getIds(); }, { tags: ["Basic", "Shared"]});
+    Opportunity.prototype.publish("currentGroup", "", "set", "label in Opportunity", function() { return this.getIds(); }, { tags: ["Basic", "Shared"]});
+    Opportunity.prototype.publish("width", 1100, "number", "label in Opportunity", function() { return this.columns(); }, { tags: ["Basic", "Shared"]});
+    Opportunity.prototype.publish("addColumn", null, "string", "number of columns in a table", function() { return this.columns(); }, { tags: ["Basic", "Shared"], 
+        editor_input: function(context, widget, cell, param) {
+            cell.append("button")
+                .attr("id", context.id() + "_addColumn" + param.id)
+                .classed("property-input custom-editor-input addColumn", true)
+                .attr("class", "update")
+                .text("AddColumn")
+                .on("click", function() {
+                    groupCount = groupCount + 1;
+                    var new_value_after_click = "Added a new column";
+                    context.setProperty(widget, param.id, new_value_after_click);
+                });
+        }
+    });
+    Opportunity.prototype.publish("removeColumn", null, "string", "number of columns in a table", function() { return this.columns(); }, { tags: ["Basic", "Shared"], 
+        editor_input: function(context, widget, cell, param) {
+            cell.append("button")
+                .attr("id", context.id() + "_removeColumn" + param.id)
+                .classed("property-input custom-editor-input removeColumn", true)
+                .text("RemoveColumn")
+                .on("click", function() {
+                    groupCount = groupCount - 1;
+                    var new_value_after_click = "Removed a column";
+                    context.setProperty(widget, param.id, new_value_after_click);
+                });
+        }
+    });
+    Opportunity.prototype.publish("headerLabels", [], "propertyArray", "Source Columns", null, { autoExpand: Column });
+    Opportunity.prototype.publish("mouseHover", [], "propertyArray", "mouse hover options", null, { autoExpand: mouseHoverColumn });
+    Opportunity.prototype.publish("columnData", [], "propertyArray", "column data", null, { autoExpand: columnDropdown });
+
+    Opportunity.prototype.enter = function(domNode, element) {
+        SVGWidget.prototype.enter.apply(this, arguments);
+        var paddingTop = 30;
+        var nodeRectHeight = 14;
+        var verticalPadding = 10;
+        var h = (this.data().length + 1) * (nodeRectHeight + verticalPadding + 1) + paddingTop;
+        this.svg = element.append("g")
+            .attr("id", "opportunityID")
```

Comment:
It is needed because simultaneously we have to regenerate opportunity svg based on id.